processing-service-container/src/test/java/com/github/cafdataprocessing/processing/service/tests/GlobalConfigIT.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2015-2017 EntIT Software LLC, a Micro Focus company.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing.processing.service.tests;
+
+import com.github.cafdataprocessing.processing.service.client.ApiClient;
+import com.github.cafdataprocessing.processing.service.client.ApiException;
+import com.github.cafdataprocessing.processing.service.client.api.GlobalConfigApi;
+import com.github.cafdataprocessing.processing.service.client.model.GlobalConfig;
+import com.github.cafdataprocessing.processing.service.client.model.GlobalConfigsEntry;
+import com.github.cafdataprocessing.processing.service.tests.utils.ApiClientProvider;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import static org.junit.Assert.*;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class GlobalConfigIT
+{
+    private static final String TEST_KEY_1 = "test_key_1";
+    private static final String TEST_DESCRIPTION_1 = "test_description_1";
+    private static final String TEST_VALUE_1 = "test_value_1";
+    private static final String TEST_NOT_EXISTING_KEY = "test_not_existing_key";
+    
+    private static GlobalConfigApi globalConfigAPI;
+    
+    @BeforeClass
+    public static void setup() throws Exception {
+        final ApiClient apiClient = ApiClientProvider.getApiClient();
+        globalConfigAPI = new GlobalConfigApi(apiClient);
+    }
+    
+    @Test
+    public void createAndGetGlobalConfig() throws ApiException {
+        createConfig("1");
+        
+        final GlobalConfig expectedConfig = new GlobalConfig();
+        expectedConfig.setDefault(TEST_VALUE_1);
+        expectedConfig.setDescription(TEST_DESCRIPTION_1);
+        
+        final GlobalConfig actualConfig = globalConfigAPI.getGlobalConfig(TEST_KEY_1);
+        assertEquals(actualConfig, expectedConfig);
+    }
+    
+    @Test
+    public void getGlobalConfigOnNotExistingRecord() throws ApiException {
+        try {
+            globalConfigAPI.getGlobalConfig(TEST_NOT_EXISTING_KEY);
+        } catch (ApiException e) {
+            assertTrue(e.getCode() == 404);
+            assertTrue(e.getMessage().contains("Failed to find global config with key '" + TEST_NOT_EXISTING_KEY + "'."));
+        }
+    }
+    
+    @Test
+    public void getAllGlobalConfigsForExistigRecords() throws ApiException {
+        createConfig("1");
+        createConfig("2");
```

Comment:
This config with "1" as the basis was created already in 'createAndGetGlobalConfig'. You should avoid conflicts where tests are modifying assets from another test. Can be done here by passing a random UUID string as the basis when calling create
e.g. createConfig("1_"+ UUID.randomUUID().toString());

Reply:
I would not change that because using "e.g. createConfig("1_"+ UUID.randomUUID().toString());" would not guarantee the size of records in store is what is expected to be. That is because an execution order of tests is not guaranteed to be the same every time they are executed an I have to consider what other tests could do. An option here is that you are fine with a check that asserts it has >1 records.

These tests are not independent units as they are working with the same store, unless you want me to remove the records after every test.

Reply:
I think the tests should be independent units so any configs created during a test should be removed on conclusion of the test.